src/app/proposals/new/review/grants/page.tsx

```diff
@@ -29,8 +32,10 @@ export default function GrantsProposalReview() {
     setLoading(true)
     try {
       if (!record?.form || record?.category !== ProposalCategory.Grants) return
-      const { description, proposalName, targetAddress, token, transferAmount, discourseLink } = record.form
-      const proposalDescription = `${proposalName};${description} ${DISCOURSE_LINK_SEPARATOR}${discourseLink} `
+      const { description, proposalName, targetAddress, token, transferAmount, discourseLink, milestone } =
+        record.form
+      const milestoneString = milestone ? `${MILESTONE_SEPARATOR + snakeCase(milestone)}; ` : ''
```

Comment:
It's redundant to have `snakeCase(milestone)` since it produces `"M1lestone:milestone_1"`. The milestone value can be simply a number, ex: `"M1lestone:1"`. It also reduces the number of characters, which is important given the smart contract's character limit.